### PR TITLE
Make the interface to choose the fonts a bit simpler.

### PR DESCRIPTION
**Pull request by boutinb**
Remove the checkbox to set the default font, and add it as default in the DropDown.
Remove warning due to form not being defined when GridLayout is used outside of an analysis form.

### CODE REVIEW

File: JASP-Desktop/gui/preferencesmodel.cpp [hunk 1]
```diff
@@ -430,14 +377,70 @@ void PreferencesModel::_loadDatabaseFont()
...
+	_allCodeFonts.removeAll(defaultCodeFont());
+	_allResultFonts.removeAll(defaultResultFont());
+	_allInterfaceFonts.removeAll(defaultInterfaceFont());
+}
 
-	emit allFontsChanged(_allFonts);
+QString PreferencesModel::_checkFontList(QString fonts) const
+{
+	if (fonts.contains(","))
+	{
```
Comment (JorisGoosen) [inline JASP-Desktop/gui/preferencesmodel.cpp]: Why all these extra braces man... Waste of space

File: Resources/Help/preferences/prefsui_nl.md [hunk 2]
```diff
@@ -5,7 +5,7 @@ Met de gebruikersinterface parameters in JASP kunnen de volgende opties gespecif
 (Al deze instellingen blijven werken, zelfs na het opnieuw opstarten van JASP.)
 
 ## Lettertype
-Hier kun je aangeven welke lettertype wordt gebruikt door de interface (file menu, ribbon bar en analysisopties paneel), de R, Lavaan of JAGS code, of de analyseresultaat paneel. Als je de standaard lettertype wilt, klik op het 'Standaard lettertype' optie.
+Hier kun je aangeven welke lettertype wordt gebruikt door de interface (file menu, ribbon bar en analysisopties paneel), de R, Lavaan of JAGS code, of de analyseresultaat paneel. Als je geen keuze maakt, de standaard lettertype wordt gebruikt.
```
Comment (JorisGoosen) [inline Resources/Help/preferences/prefsui_nl.md]: Dit is geen correct nederlands...

"Hier kun je aangeven welk lettertype gebruikt wordt door de interface (bestandsmenu, ribbon balk en de analyseopties), code invoer, of de resultaten. Als je geen keuze maakt, wordt het standaard lettertype gebruikt."

**Review by JorisGoosen — CHANGES_REQUESTED** [review]
Why did you simplify it anyway?

For me it is now **very hard** to get back to the "default" font:
Screenshot_2020-08-31_141014

I can also not type in the box, there is no scrollbar so I can only scroll with the mouse but that goes very slow...
I have a lot of fonts on my system apparently...

For me the "default" button was good, at least you can go back to default easily.

**Comment by TimKDJ** [comment]
This is quite a bit more clean and it's also more clear what is going on.
However, I think it'd be good to:
- [ ]  Not make the default gray; it's unclear you can click this again to go back to your default
- [ ] Add a scrollbar, as mentioned by Joris
- [ ] Add a separator line between the default and the other fonts

I think this is done in a very neat way:
chose-font

**Comment by JorisGoosen** [comment]
Looking good Bruno!

But you should turn on clipping for the dropdown menu (or set the size right):
afbeelding

**Comment by JorisGoosen** [comment]
^^
afbeelding

**Pull request merged**